### PR TITLE
Bluetooth initialization order bug fix (and housekeeping)

### PR DESCRIPTION
**Pull request by scfrontier**
When a baud rate change was requested, we were trying to connect at the new baud rate instead of the current "working" baud rate. This was reporting back to the console as a NAME change failure.

Also moved some code to create a "findBaudIndex" function. This is not necessary but helped me keep things organized.

### CODE REVIEW

File: firmware/console/binary/bluetooth.cpp [hunk 1]
```diff
@@ -267,6 +230,48 @@ static void runCommands(SerialTsChannelBase* tsChannel) {
...
+		tsChannel->stop();
+		chThdSleepMilliseconds(10);	// safety
+
+		if (baudIdx == efi::size(baudRates)) {
+			efiPrintf("Failed to find current BT module baudrate");
+			tsChannel->start(engineConfiguration->tunerStudioSerialSpeed);
+			return 255;//failed
+		}
+
+		efiPrintf("Restarting at %d", baudRates[baudIdx].rate);
```
Comment (dron0gus) [inline firmware/console/binary/bluetooth.cpp:245]: %lu instead of %d

**Review by dron0gus — CHANGES_REQUESTED** [review]

**Comment by rusefillc** [comment]
@scfrontier is there any relation to https://github.com/rusefi/rusefi/pull/6620?

**Comment by scfrontier** [comment]
I'm not sure about [#6620](https://github.com/rusefi/rusefi/pull/6620), it might be a coincidence, or maybe they saw my post on the forum https://rusefi.com/forum/viewtopic.php?f=4&t=340&p=49653#p49653 and are trying to help.

I am trying to figure out their code changes. I will copy jcor25's code and see what happens.

**Comment by rusefillc** [comment]
@scfrontier there is some life on the discord server https://github.com/rusefi/rusefi/wiki/Discord

**Comment by rusefillc** [comment]
@scfrontier there is also the part that #6620 is not in a good place as a PR with all the unrelated changes. Any chance you could decide on best approach in a blean PR?

**Comment by scfrontier** [comment]
I have not been able to reach jcor on Discord for discussion. This pull request has fewer unrelated changes and fewer file changes. I cannot provide evidence as to which solution is best as they both solve the problem in different ways.

**Pull request merged**